tests/handlers/test_typing.py

```diff
@@ -98,7 +99,13 @@ def prepare(self, reactor: MemoryReactor, clock: Clock, hs: HomeServer) -> None:
         mock_notifier = hs.get_notifier()
         self.on_new_event = mock_notifier.on_new_event
 
-        self.handler = hs.get_typing_handler()
+        self.handler = cast(TypingWriterHandler, hs.get_typing_handler())
+
+        # hs.get_typing_handler will return a TypingWriterHandler when calling it
+        # from the main process, and a FollowerTypingHandler on workers.
+        # We rely on methods only available on the former, so assert we have the
+        # correct type here.
+        self.assertIsInstance(self.handler, TypingWriterHandler)
```

Comment:
We should just be able to do the `assert` without the `cast`; the trick is you need to use a bare `assert` or mypy won't realize it is being used for type narrowing.

```python
handler = hs.get_typing_handler()
assert isinstance(handler, TypingWriterHandler)
self.handler = handler

```

Reply:
Nice trick. So mypy needs to have the assert before you assign the type of the variable for the first time?

Reply:
Only because you're using it in other scopes. Otherwise it thinks the property in the class is of the wrong type and you need to assert each time you use it.